[refactor](planner) In the runtime filter, use crc_hash replace original murmur_hash

Mryange opened the pull request:
# Proposed changes

After testing, crchash is 4 to 5 times faster than murmurhash.

## Problem summary

When be_exec_version is 1, the original murmurhash method is still used. If updated, the new crchash method will be used.

Describe your changes.


## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

Mryange committed:
Delete the environment variable enable_vectorized_engine

Mryange committed:
Delete the environment variable enable_vectorized_engine

Mryange committed:
rebase

Mryange committed:
rmvec5

Mryange committed:
rmvec6

Mryange committed:
use new hash

Mryange commented:
run buildall

HappenLee commented on be/src/exprs/runtime_filter.cpp:
```diff
@@ -1011,6 +1017,11 @@ class RuntimePredicateWrapper {
+
+    // When _use_new_hash is set to true, use the new hash method. 
+    // This is only to be used if the be_exec_version may be less than 2. If updated, please delete it.
+    bool _use_new_hash;
```
const bool

HappenLee commented on be/src/vec/exprs/vbloom_predicate.cpp:
```diff
@@ -67,10 +67,21 @@ Status VBloomPredicate::execute(VExprContext* context, Block* block, int* result
+                const StringRef v(ele.data, ele.size);
+                ptr[i] = _filter->find_new_hash(reinterpret_cast<const void*>(&v));
+            }
+        }else {
```
code format

HappenLee replied:
code format

Mryange committed:
format

Mryange commented:
run buildall

BiteTheDDDDt commented on be/src/vec/exprs/vbloom_predicate.cpp:
```diff
@@ -67,10 +67,20 @@ Status VBloomPredicate::execute(VExprContext* context, Block* block, int* result
     auto ptr = ((ColumnVector<UInt8>*)res_data_column.get())->get_data().data();
     auto type = WhichDataType(remove_nullable(block->get_by_position(arguments[0]).type));
     if (type.is_string_or_fixed_string()) {
```
Why the new hash only used at string type?

hello-stephen commented:
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.54 seconds
 stream load tsv:          450 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230404090728_clickbench_pr_124614.html

Mryange commented:
run buildall

AshinGau committed:
[fix](file_reader) bad_typeid when reading csv&json files (#18400)

PR(#18340) resolve the conflict with PR(#18301) has changed the file_reader to create, resulting in e: [E-123] std::bad_typeid exception.

ByteYue committed:
[chore](regression) prevent creating stmt failed in cold heat separation regression case #18391

Previously in cold_heat_separation regression, it just tries to create resources/policies. Sometimes if the former cases failed or BE crashed when doing cases the resources would not be cleared so the next time invoking this regression cases would result in failure.
